src/puppetlabs/trapperkeeper/logging.clj

```diff
@@ -72,25 +72,26 @@
             (.toInt level))
        (.setLevel root level)))))
 
-(defn configure-logger-via-file!
-  "Reconfigures the current logger based on the supplied configuration
-  file."
-  [logging-conf-file]
-  {:pre [(string? logging-conf-file)]}
+(defn configure-logger!
+  "Reconfigures the current logger based on the supplied configuration."
+  [logging-conf]
+  {:pre [(#{String java.io.File java.net.URL java.io.InputStream org.xml.sax.InputSource} (type logging-conf))]}
```

Comment:
I don't think pre-conditions should artificially restrict what values a function can handle.  In this case, it feels too onerous to bother with pre-conditions, I think it'd be best to just delete it altogether.